portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Task;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/data")
+public final class DataServlet extends HttpServlet {
+    // Set default to 5
+
+    int maxCommentsPosted = 5;
+	@Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+    
+        // Gather all the user comments and sort them by timestamp
+        Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Task> commentsRecord = new ArrayList<>();
+        int count = 0;
+        for (Entity entity : results.asIterable()) {
+            long id = entity.getKey().getId();
+            String username = (String) entity.getProperty("userName");
+            String comment = (String) entity.getProperty("userComment");
+            long timestamp = (long) entity.getProperty("timestamp");    
+            // Limit amount of comments that will be shown to 4
+            count++;    
+            if(count <= maxCommentsPosted){
+                Task userEntry = new Task(id, username, comment, timestamp);
+            	commentsRecord.add(userEntry);
+            }
+            
+    	}
```

Comment:
What is the goal of this code? Please add documentation here so I can understand.